From 234780ef33626acb37cd2e771d0839f36894c908 Mon Sep 17 00:00:00 2001
From: Jordan R Abrahams-Whitehead <ajordanr@google.com>
Date: Wed, 26 Feb 2025 19:25:38 +0000
Subject: [PATCH] Revert "[lldb] Add terminfo dependency for ncurses support"

This reverts commit 41875593066a603ff4cd746a430e481fd7d3a180.

This commit is no longer needed given upstream has merged
a proper fix with commits:

[1] 8fff0c181f26a5e8b2344c061ebf2559118b1160
[2] bb6a273d9ab9ee90dbb957e541f4d810fffb22ee

patch.cherry: false
patch.platforms: chromiumos
patch.version_range.from: 537678
patch.version_range.until: 566148

---
 lldb/cmake/modules/FindCursesAndPanel.cmake | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/lldb/cmake/modules/FindCursesAndPanel.cmake b/lldb/cmake/modules/FindCursesAndPanel.cmake
index cd2a68a01a0d..aaadf214bf54 100644
--- a/lldb/cmake/modules/FindCursesAndPanel.cmake
+++ b/lldb/cmake/modules/FindCursesAndPanel.cmake
@@ -2,15 +2,12 @@
 # FindCursesAndPanel
 # -----------
 #
-# Find the curses, terminfo, and panel library as a whole.
-# NOTE: This is patched due to a terminfo dependency error introduced in
-# https://github.com/llvm/llvm-project/pull/92865
+# Find the curses and panel library as a whole.
 
-if(CURSES_INCLUDE_DIRS AND CURSES_LIBRARIES AND TINFO_LIBRARIES AND PANEL_LIBRARIES)
+if(CURSES_INCLUDE_DIRS AND CURSES_LIBRARIES AND PANEL_LIBRARIES)
   set(CURSESANDPANEL_FOUND TRUE)
 else()
   find_package(Curses QUIET)
-  find_package(TINFO_LIBRARIES NAMES tinfo DOC "The curses tinfo library" QUIET)
   find_library(PANEL_LIBRARIES NAMES panel DOC "The curses panel library" QUIET)
   include(FindPackageHandleStandardArgs)
   find_package_handle_standard_args(CursesAndPanel
@@ -19,10 +16,9 @@ else()
                                     REQUIRED_VARS
                                       CURSES_INCLUDE_DIRS
                                       CURSES_LIBRARIES
-                                      TINFO_LIBRARIES
                                       PANEL_LIBRARIES)
-  if(CURSES_FOUND AND TINFO_LIBRARIES AND PANEL_LIBRARIES)
-    mark_as_advanced(CURSES_INCLUDE_DIRS CURSES_LIBRARIES TINFO_LIBRARIES PANEL_LIBRARIES)
+  if(CURSES_FOUND AND PANEL_LIBRARIES)
+    mark_as_advanced(CURSES_INCLUDE_DIRS CURSES_LIBRARIES PANEL_LIBRARIES)
   endif()
 endif()
 
-- 
2.49.0.395.g12beb8f557-goog